contracts/SponsorFaucet.sol

```diff
@@ -29,35 +29,18 @@ contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard, WhitelistedRole {
         uint256 upper_bound;
     }
 
-    /*** bounds set by foundation ***/
-    //small quota
-    //total sponsored limit
-    uint256 public gas_total_limit;
-    uint256 public collateral_total_limit;
-    //single sponsor bound
-    uint256 public gas_bound;
-    uint256 public collateral_bound;
-    //upper bound for single tx
-    uint256 public upper_bound;
-
-    // general quota
-    //total sponsored limit
-    uint256 public general_gas_total_limit;
-    uint256 public general_collateral_total_limit;
-    //single sponsor bound
-    uint256 public general_gas_bound;
-    uint256 public general_collateral_bound;
-    //upper bound for single tx
-    uint256 public general_upper_bound;
+    //constant address used as key for small and large
+    address constant small = 0x0000000000000000000000000000000000000000;
+    address constant large = 0x0000000000000000000000000000000000000001;
 
     // for all contracts
     mapping(address => detail) public dapps;
     
-    // bounds for special contracts
-    mapping(address => bounds) public special_dapps;
+    // bounds for contracts
+    mapping(address => bounds) public dapp_bounds;
 
-    address[] public general_contracts;
-    address[] public special_contracts;
+    mapping(address => bool) public large_contracts;
+    mapping(address => bool) public custom_contracts;
```

Comment:
It seems `dapp_bounds` is enough to check whether contract has customized sponsorship.

Reply:
yes, but **large** and **small** will be mixed if use dapp_bounds only.